### PR TITLE
Fix/ie11

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/9242

All our code is transpiled by babel to be able to run in ie11.

But some of our dependencies have opted to no longer support ie11.

Because of this we need to transpile those dependencies AGAIN, to make them compatible.

We want to attempt to do as much of this transpilation ahead of time, so when the user starts storybook, their machine doesn't have to do unnecessary work.

I'm trying to do this work in the DLL plugin (which is a pre-compilation step we do, so webpack doesn't have to traverse the entire dependency graph when creating the bundle).

## What I did

- ADD babel loader for all modules passed when creating the manager's DLL.
- ADD a DLL for the docs addon
- ADD preset for graphql addon so it adds a babel loader
- ADD dedicated template file for virtual modules

## How to test

Run storybook locally, connect to browserstack

### CODE REVIEW

**Comment by ndelangen** [comment]
ALSO:

presets can now export a single function, which will get called with storybookOption, and can return an object with presetProperties.

@shilman This is a much cleaner solution for docs preset. 😘

**Comment by tooppaaa** [comment]
Fuse issue has been raised https://github.com/krisk/Fuse/issues/374

File: lib/core/src/server/preview/iframe-webpack.config.js [hunk 1]
```diff
@@ -52,18 +54,25 @@ export default ({
...
+        if (m.decorators) {
+          m.decorators.forEach(function(decorator){ return addDecorator(decorator); });
+        }
+        if (m.parameters || m.globalArgs || m.globalArgTypes) {
+          addParameters(Object.assign({}, m.parameters, { globalArgs: m.globalArgs, globalArgTypes: m.globalArgTypes }));
+        }
+        if (m.parameterEnhancers) {
+          m.parameterEnhancers.forEach(function(enhancer){ return addParameterEnhancer(enhancer)});
+        }
       `;
```
Comment (ndelangen) [inline lib/core/src/server/preview/iframe-webpack.config.js:76]: This code should be in a separate file.

If we do something like this:
```
      virtualModuleMapping[entryFilename] = require('fs-extra')
        .readFileSync('./virtualModule.js')
        .replace('__FOO__BAR__', JSON.stringify(configFilename));
```

We can not only get syntax highlighting, eslint, autocomplete etc on it.. but we can also get to use ES6 in this file.

File: lib/core/src/server/preview/iframe-webpack.config.js [hunk 2]
```diff
@@ -80,7 +89,7 @@ export default ({
   return {
     mode: isProd ? 'production' : 'development',
     bail: isProd,
-    devtool: '#cheap-module-source-map',
+    devtool: 'none',
```
Comment (ndelangen) [inline lib/core/src/server/preview/iframe-webpack.config.js:92]: should be reverted after

File: lib/core/src/server/preview/iframe-webpack.config.js (hunk 1 above)
Comment (shilman) [reply lib/core/src/server/preview/iframe-webpack.config.js:76]: Great idea ❤️

Comment (tooppaaa) [reply lib/core/src/server/preview/iframe-webpack.config.js]: Done, I let you check !

**Comment by ndelangen** [comment]
I think it's ready

File: examples/official-storybook/package.json [hunk 3]
```diff
@@ -3,8 +3,8 @@
   "version": "6.0.0-alpha.30",
   "private": true,
   "scripts": {
-    "build-storybook": "cross-env STORYBOOK_DISPLAY_WARNING=true DISPLAY_WARNING=true build-storybook -c ./",
-    "debug": "cross-env NODE_OPTIONS=--inspect-brk STORYBOOK_DISPLAY_WARNING=true DISPLAY_WARNING=true start-storybook -p 9011 -c ./ --no-dll",
+    "build-storybook": "cross-env STORYBOOK_DISPLAY_WARNING=true DISPLAY_WARNING=true build-storybook -c ./ --no-dll",
```
Comment (tooppaaa) [inline examples/official-storybook/package.json:6]: Is this intended ?

Comment (ndelangen) [reply examples/official-storybook/package.json:6]: 👍

**Review by tooppaaa — APPROVED** [review]

**Comment by afebbraro** [comment]
Hey @shilman , I'm not able to get storybook 5.3.18 to load in IE11 with Angular 9. Could this be related?

**Comment by shilman** [comment]
@afebbraro most of these changes are related to new versions of storybook dependencies that are not IE11 compatible. i'm not sure how many of them are present in 5.3.x vs 6.0 -- maybe @tooppaaa or @ndelangen can shed some light. if we can make small changes to fix 5.3 we will, otherwise you'll have to wait for 6.0. any idea what's the most recent 5.x version that's worked for you?

**Comment by afebbraro** [comment]
hey @shilman  thanks for the quick response!! We recently upgraded to Angular 9.1 from 8.2 and thats when the problem started. I went back each patch version all the way to 5.3.14 with Angular 9.1 and it wont load in IE11. "syntax error". When were you thinking 6.0 will be released as the stable version (roughly)?

Here's my package.json!!
```
 "dependencies": {
    "@angular/animations": "~9.1.0",
    "@angular/common": "~9.1.0",
    "@angular/compiler": "~9.1.0",
    "@angular/core": "~9.1.0",
    "@angular/forms": "~9.1.0",
    "@angular/platform-browser": "~9.1.0",
    "@angular/platform-browser-dynamic": "~9.1.0",
    "@angular/router": "~9.1.0",
    "classlist.js": "^1.1.20150312",
    "replace": "^1.1.1",
    "rxjs": "~6.5.4",
    "tiny-date-picker": "github:sparkdesignsystem/tiny-date-picker#v3.2.9",
    "tslib": "^1.10.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.901.0",
    "@angular-devkit/build-ng-packagr": "~0.901.0",
    "@angular/cli": "~9.1.0",
    "@angular/compiler-cli": "~9.1.0",
    "@angular/language-service": "~9.1.0",
    "@babel/preset-react": "^7.7.4",
    "@babel/traverse": "^7.6.0",
    "@compodoc/compodoc": "^1.1.11",
    "@storybook/addon-a11y": "^5.3.18",
    "@storybook/addon-docs": "^5.3.18",
    "@storybook/addon-google-analytics": "^5.3.18",
    "@storybook/addon-jest": "^5.3.18",
    "@storybook/addons": "^5.3.18",
    "@storybook/angular": "^5.3.18",
    "@storybook/preset-scss": "^1.0.2",
    "@storybook/theming": "^5.3.18",
    "@types/jest": "^25.1.3",
    "@types/node": "^12.11.1",
    "babel-loader": "^8.0.6",
    "codelyzer": "^5.1.2",
    "css-loader": "^3.1.0",
    "eslint": "^6.5.1",
    "jest": "^25.1.0",
    "jest-preset-angular": "^8.1.2",
    "ng-packagr": "^9.0.0",
    "node-sass": "^4.13.1",
    "protractor": "~5.4.0",
    "sass-loader": "^7.1.0",
    "style-loader": "^0.23.1",
    "ts-node": "~7.0.0",
    "tslint": "~5.15.0",
    "typescript": "^3.6.4",
    "whatwg-fetch": "^3.0.0"
  }
```

Thanks again!

**Comment by shilman** [comment]
Shooting for 6.0 stable in mid-May, but it's a big one so ... 🤞

**Comment by jeffbelser-okta** [comment]
@shilman @ndelangen Thanks for your work here

Our team would love to have this fix released as part of the 5.x effort as we can't currently access IE11 from the latest release! Would this be possible as a new minor release?

**Comment by shilman** [comment]
@jeffbelser-okta Unfortunately we can't release these changes as a minor. 6.0 should RC next week.

**Pull request merged**